pkg/analyzer/lib/src/generated/gn.dart

```diff
@@ -243,26 +155,56 @@ class GnWorkspace extends Workspace {
     return null;
   }
 
-  /**
-   * Locate the Dart sources directory.
-   *
-   * Return `null` if it could not be found.
-   */
-  static String _getPackagesDirectory(ResourceProvider provider, String root) {
+  static String _findPackagesFile(
```

Comment:
Is there some documentation we can reference here that explains the structure that's being traversed, why there are (or can be) multiple .packages files, and why it's reasonable to choose one at random? I'm concerned about having to maintain this code when I don't understand why it works the way it does.